### PR TITLE
Update NOTES.md

### PR DESCRIPTION
**Pull request by raymond-chetty**
#531: Quick document update about UID & GID using 1000 in devcontainer base images.

### CODE REVIEW

**Comment by raymond-chetty** [comment]
Its possible to resolve this if we add a configuration to usermod a username as well as UID & GID. Possibly open a ticket for that.

File: src/common-utils/NOTES.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 This Feature should work on recent versions of Debian/Ubuntu, RedHat Enterprise Linux, Fedora, RockyLinux, and Alpine Linux.
 
+## Base Images
```
Comment (samruddhikhandale) [inline src/common-utils/NOTES.md:5]:
```suggestion
## Using with dev container images
```

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]
Appreciate your help in improving our documentation, thank you! 

Left some minor comments.

**Comment by samruddhikhandale** [comment]
> Looking at how the vscode wizard works, you can't manually type in a UID value only select from an option,

As mentioned in https://github.com/devcontainers/features/issues/531#issuecomment-1524168766,  I believe that's a bug with the wizard. Feel free to open an issue in https://github.com/microsoft/vscode-remote-release/issues

> Looking at how the vscode wizard works, you can't manually type in a UID value only select from an option, it might be better to assume that if UID 1000 exists it belongs to the base image user and that setting changes should all be applied to UID 1000, possibly including UID changes.

> Logic:
> if UID 1000 exists -> mod the user
> if UID X exists -> mod the user (no UID changes)
> else add the user
> 
> This way wizard users will do what they'd expect, modify the base user, assuming we leave the UID & GID 1000 option. Also, manual configurations will be possible with a variety of scenarios either adding or updating a user.

Thanks for sharing your thoughts, it's really helpful. 

We could make this change to the common-utils Feature, however, I am not sure if the logic could become confusing for users ; instead I think a clear error message could navigate them to use different UID/GID to unblock them.
I am happy to help out with the change if people think otherwise.

File: src/common-utils/NOTES.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
...
+
+This feature is used in many of the devcontainer [base images](https://github.com/devcontainers/images), as a result
+those base images have already allocated UID & GID 1000. Attempting to add this feature on a new configuration utilizing
+a devcontainer base image may result in an error when building if adding a new user and assiging UID or GID to 1000. 
+
```
[the remark this answers is not in the document]
Comment (raymond-chetty) [reply src/common-utils/NOTES.md:10]: Thank you for your suggestion. I approve of this correction on behalf of the people of the united states. @PresidentObama :)

File: src/common-utils/NOTES.md (hunk 1 above)
Comment (raymond-chetty) [reply src/common-utils/NOTES.md:5]: approved!

Comment (samruddhikhandale) [reply src/common-utils/NOTES.md:5]: Same here!

**Comment by nanis94** [comment]
devcontainers:main

**Comment by nanis94** [comment]
devcontainers:main

**Comment by nanis94** [comment]
devcontainers:main

**Review by nanis94 — COMMENTED** [review]
devcontainers:main

**Review by samruddhikhandale — APPROVED** [review]
Thanks!

**Pull request merged**